### PR TITLE
Use sizeof(field) instead of sizeof(type) and other fixes.

### PR DESCRIPTION
**Pull request by robertnishihara**

### CODE REVIEW

File: src/common/state/redis.c [hunk 1]
```diff
@@ -254,8 +250,9 @@ void redis_result_table_add(table_callback_data *callback_data) {
   /* Add the result entry to the result table. */
   int status = redisAsyncCommand(db->context, redis_result_table_add_callback,
                                  (void *) callback_data->timer_id,
-                                 "SET result:%b %b", id.id, sizeof(object_id),
-                                 (*result_task_id).id, sizeof(task_id));
+                                 "SET result:%b %b", id.id, sizeof(id.id),
+                                 (*result_task_id).id,
+                                 sizeof((*result_task_id).id));
```
Comment (mehrdadn) [inline src/common/state/redis.c]: nit (sorry): `result_task_id->id`

File: src/common/state/redis.c [hunk 2]
```diff
@@ -466,8 +463,8 @@ void redis_task_table_get_task(table_callback_data *callback_data) {
   int status =
       redisAsyncCommand(db->context, redis_task_table_get_task_callback,
                         (void *) callback_data->timer_id, "HGETALL task:%b",
-                        id.id, sizeof(task_id));
-  if ((status == REDIS_ERR) || db->sub_context->err) {
+                        id.id, sizeof(id.id));
+  if ((status == REDIS_ERR) || db->context->err) {
```
Comment (mehrdadn) [inline src/common/state/redis.c]: This change seems a little out of the context (heh) of the rest of this commit... was it intentional?

File: src/global_scheduler/global_scheduler_algorithm.c [hunk 3]
```diff
@@ -24,6 +24,7 @@ void handle_local_scheduler_heartbeat(global_scheduler_state *state) {
 
 void handle_new_local_scheduler(global_scheduler_state *state,
                                 db_client_id db_client_id) {
-  local_scheduler local_scheduler = {.id = db_client_id};
+  local_scheduler local_scheduler;
+  local_scheduler.id = db_client_id;
```
Comment (mehrdadn) [inline src/global_scheduler/global_scheduler_algorithm.c]: Sorry to pick on this again, but you can actually just shorten this to `local_scheduler local_scheduler = {db_client_id};`.
Also, be aware that there's a slight difference between your new version and using brace initialization -- if any fields are added later, they'll be zero-initialized when you use brace initialization, but they'll be uninitialized in your version. So if you're trying to initialize the whole struct, use braces.

File: src/common/state/redis.c (hunk 2 above)
Comment (robertnishihara) [inline src/common/state/redis.c]: I think the `sub_context` -> `context` is a bug fix.

File: src/plasma/plasma_manager.c [hunk 4]
```diff
@@ -151,8 +151,8 @@ int send_client_reply(client_connection *conn, plasma_reply *reply) {
   CHECK(conn->num_return_objects >= 0);
   --conn->num_return_objects;
   /* TODO(swang): Handle errors in write. */
-  int n = write(conn->fd, (uint8_t *) reply, sizeof(plasma_reply));
-  return (n != sizeof(plasma_reply));
+  int n = write(conn->fd, (uint8_t *) reply, sizeof(reply));
```
Comment (mehrdadn) [inline src/plasma/plasma_manager.c]: `sizeof(*reply)`?

File: src/common/state/redis.c (hunk 2 above)
Comment (mehrdadn) [inline src/common/state/redis.c]: Oh ok cool.

File: src/global_scheduler/global_scheduler_algorithm.c (hunk 3 above)
Comment (robertnishihara) [inline src/global_scheduler/global_scheduler_algorithm.c]: Good point. I've had trouble with initializations like `local_scheduler = {db_client_id};` in the past when I add new fields (or change the order of fields) because then sometimes the entries in the braces end up corresponding to the wrong fields.. do you know a good way to avoid this?

File: src/plasma/plasma_manager.c (hunk 4 above)
Comment (robertnishihara) [inline src/plasma/plasma_manager.c]: Ok, that was the bug. Good catch!

File: src/global_scheduler/global_scheduler_algorithm.c (hunk 3 above)
Comment (robertnishihara) [inline src/global_scheduler/global_scheduler_algorithm.c]: How about `memcpy(..., 0, ...)` followed by setting the fields manually?

Comment (mehrdadn) [inline src/global_scheduler/global_scheduler_algorithm.c]: Ah I see, I didn't realize the field orders were changing often. In that case, if you do want the zero-initialization, then in lieu of the braces you can do `memset(&local_scheduler, 0, sizeof(local_scheduler))` and then just assign to the field like you have here. If you don't, then just keep what you have as is.

File: src/plasma/plasma_manager.c [hunk 5]
```diff
@@ -151,8 +151,8 @@ int send_client_reply(client_connection *conn, plasma_reply *reply) {
...
+  return (n != sizeof(reply));
```
Comment (mehrdadn) [inline src/plasma/plasma_manager.c]: Fix this line too?

File: src/plasma/plasma_manager.c (hunk 4 above)
Comment (mehrdadn) [inline src/plasma/plasma_manager.c]: Yay! But see below.

**Comment by mehrdadn** [comment]
(lgtm)

**Comment by robertnishihara** [comment]
This is currently failing the valgrind tests. It may be the switch from signaling to atexit.

**Comment by mehrdadn** [comment]
Hm, that's unfortunate. If it works fine with signaling then I guess switch back and I'll hopefully try to take a look and see why `atexit` is failing if/when I get the chance.

**Comment by robertnishihara** [comment]
Yeah, assuming that is in fact the problem, I'm sure it can be made to work. Will just need to look into it more.

File: src/common/state/redis.c (hunk 2 above)
Comment (pcmoritz) [inline src/common/state/redis.c]: It also needs to be swapped in the line after
